fix(Call n8n Workflow Tool Node): Return all items from subexecution

mutdmour opened the pull request:
## Summary
Return all items from first run of first subexecution
<img width="1323" alt="Screenshot 2025-02-20 at 14 48 21" src="https://github.com/user-attachments/assets/9dbf4b5f-892d-42f5-bcd1-9d6e9f2537e1" />

<!--
Describe what the PR does and how to test.
Photos and videos are recommended.
-->

## Related Linear tickets, Github issues, and Community forum posts

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
-->
<!-- Use "closes #<issue-number>", "fixes #<issue-number>", or "resolves #<issue-number>" to automatically close issues when the PR is merged. -->


## Review / Merge checklist

- [X] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [X] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

mutdmour committed:
feat: support returning all items in service

mutdmour committed:
fix: fix up tool workflow

mutdmour committed:
fix: return data as object, add tests

mutdmour committed:
fix: types

OlegIvaniv commented on packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/ToolWorkflow.node.ts:
```diff
@@ -36,6 +37,7 @@ export class ToolWorkflow extends VersionedNodeType {
 			1.2: new ToolWorkflowV1(baseDescription),
 			1.3: new ToolWorkflowV1(baseDescription),
 			2: new ToolWorkflowV2(baseDescription),
+			2.1: new ToolWorkflowV2_1(baseDescription),
```
Do we need a separate module for this? We could just set `returnAllItems` to true if `this.getNode().typeVersion > 2.0`

mutdmour replied:
I want to avoid doing this in workflowtoolservice, but now realizing could have done this at the module level. Will refactor.

OlegIvaniv commented on packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/utils/WorkflowToolService.ts:
```diff
@@ -54,23 +53,25 @@ export class WorkflowToolService {
 		name: string;
 		description: string;
 		itemIndex: number;
+		returnAllItems?: boolean;
```
Nitpick but IMO it would be nicer to have this as an `options` property in constructor since only `executeSubWorkflow` needs the value. Now we're just passing it through in `createTool` and `runFunction`.

mutdmour reviewed (commented):
@OlegIvaniv  Thanks for the feedback

mutdmour committed:
fix: address feedback

mutdmour committed:
Merge branch 'master' of github.com:n8n-io/n8n into ado-3101-1

mutdmour committed:
fix: update tests

mutdmour commented:
@OlegIvaniv Addressed your feedback and tested again. Let me know if any other changes are needed 🙏🏽
<img width="1100" alt="Screenshot 2025-02-25 at 13 21 26" src="https://github.com/user-attachments/assets/eb910bd7-baa8-4e97-be4a-d09a74da16bc" />

OlegIvaniv reviewed (dismissed):
Thanks for addressing my points, LGTM!

mutdmour committed:
fix: Return items as items, not as list of items

mutdmour committed:
fix: address conflicts

mutdmour committed:
Merge branch 'master' of github.com:n8n-io/n8n into ado-3101-1

mutdmour committed:
fix: address conflicts

mutdmour committed:
fix: tests

mutdmour committed:
remove conflict

CharlieKolb commented on packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/v2/utils/WorkflowToolService.ts:
```diff
@@ -74,10 +83,25 @@ export class WorkflowToolService {
 				inputData: [[{ json: { query } }]],
 			});
+
+			const { index } = context.addInputData(NodeConnectionType.AiTool, [[{ json: { query } }]]);
```
the `cloneWith` call now handles adding the inputData fyi, this would add it twice.

mutdmour replied:
ah. merge issue. thanks for catching

mutdmour committed:
fix: remove input data adding

janober commented:
Got released with [`n8n@1.83.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.83.0)

salomaoparkour commented:
That's interesting, @mutdmour 
Does that mean we won't have to aggregate data in subworkflows anymore? To make sure the AI receives everything

mutdmour commented:
yes that's correct @salomaoparkour
